### PR TITLE
[MM-18361] Style text inputs and post options with theme

### PR DESCRIPTION
**Pull request by migbot**
#### Summary
Use `theme.centerChannelColor` for text/placeholder color and `theme.centerChannelBg` for background color.

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on:
* iOS simulator, 12.3
* Android emulator, 8.1

#### Screenshots
Before:
before-1
before-2

After:
after-1
after-2

### CODE REVIEW

**Comment by esethna** [comment]
Needs to also apply to the create channel dialogs (public/private). We can use other text boxes in edit profile etc for styling reference

**Comment by esethna** [comment]
Also needs a separator line on archived channels above the text box

**Comment by enahum** [comment]
@migbot post screenshots dude ;)

**Comment by enahum** [comment]
@migbot what about the reaction list, same component basically

**Comment by enahum** [comment]
@migbot you missed the indicator bar

**Comment by lindalumitchell** [comment]
Is there a ticket for this, with a list of the changes to verify? :)

**Comment by enahum** [comment]
@lindalumitchell @esethna will create a bunch of tickets of improvements that we found during the offsite including this one

**Comment by esethna** [comment]
cc// @migbot @enahum ^

**Comment by enahum** [comment]
@esethna I’ll trigger the build

**Comment by lindalumitchell** [comment]
Okay thanks EN; @esethna after there's a ticket for this one (and what all we're verifying), if you mention me here I can come back and do some testing on it (or find someone who can :) ).

**Comment by enahum** [comment]
@esethna

1. The buffer should not be even there IMO

2. https://github.com/mattermost/mattermost-mobile/pull/3221 PR for that is already submitted and it goes into 1.23

3. Can be a different PR as there is also an alignment issue.

So basically the one thing is 1.

Miguel is out till monday so I can make the changes once I finish with the other ones for the sidebar

**Comment by esethna** [comment]
Ya that's fine. For #1 though, there should be a slight buffer. See examples in the settings menu for other screens

**Comment by enahum** [comment]
@esethna archived channels look like this, I think @asaadmahmood made that change if I'm correct

image

PR Updated with the changes

File: app/screens/post_options/post_options.js [hunk 1]
```diff
@@ -66,181 +67,162 @@ export default class PostOptions extends PureComponent {
...
-                    text={formatMessage({id: 'mobile.post_info.unflag', defaultMessage: 'Unflag'})}
-                    onPress={this.handleUnflagPost}
-                    isLandscape={isLandscape}
-                />
-            );
+            key = 'unflag';
+            message = {id: t('mobile.post_info.unflag'), defaultMessage: 'Unflag'};
+            onPress = this.handleUnflagPost;
+        } else {
+            key = 'flagged';
```
Comment (sudheerDev) [inline app/screens/post_options/post_options.js]: We could assign these as the default values.

File: app/components/edit_channel_info/edit_channel_info.js [hunk 2]
```diff
@@ -249,37 +231,7 @@ export default class EditChannelInfo extends PureComponent {
...
-                                            placeholderTextColor={changeOpacity('#000', 0.5)}
-                                            underlineColorAndroid='transparent'
-                                            disableFullscreenUI={true}
-                                            keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
-                                        />
-                                    </View>
-                                </View>
-                            )}
-                            {/*TODO: Hide channel url field until it's added to CreateChannel */}
-                            {false && editing && !displayHeaderOnly && (
```
Comment (sudheerDev) [inline app/components/edit_channel_info/edit_channel_info.js]: huh weird we had left all this time

**Comment by ogi-m** [comment]
Tested on Android and looks good to merge 👍 
- Tested on Mattermost and Windows Dark themes and checked that Light theme still looks as expected
- checked text input boxes on post input area, RHS input area, create/edit channel screens, account settings inputs, search, jump-to search, emoji search
- checked long press on post and long press on reaction menus

**Comment by jgilliam17** [comment]
Tested on iOS, also looks good to merge.
- Verified both dark and light themes
- Checked text inputs, RHS text input, emoji search, long press on posts and reactions
- Checked edit and create channel and select team screens, account settings, emoji search, archived channels, jump-to search

**Pull request merged**